pkg/gds/admin.go

```diff
@@ -604,19 +606,20 @@ func (s *Admin) ListVASPs(c *gin.Context) {
 	// Query the list of VASPs from the data store
 	iter := s.db.ListVASPs()
 	defer iter.Release()
-	for iter.Next() {
-		out.Count++
+	for out.Count = 0; iter.Next(); out.Count++ {
 		if out.Count >= minIndex && out.Count < maxIndex {
 			// In the page range so add to the list reply
 			// Fetch VASP from the database
 			var vasp *pb.VASP
 			if vasp = iter.VASP(); vasp == nil {
 				// VASP could not be parsed; error logged in VASP() method continue iteration
+				out.Count--
 				continue
 			}
 
-			// Check the status before continuing
-			if status != pb.VerificationState_NO_VERIFICATION && vasp.VerificationStatus != status {
+			// Check against the status filters before continuing
+			if _, ok := filters[vasp.VerificationStatus]; len(filters) > 0 && !ok {
+				out.Count--
```

Comment:
Count is definitely still having trouble … decrementing here if we're filtering so the count reflects the number we're expecting back. Will need to write tests for this.